pkg/destroy/aws/aws.go

```diff
@@ -154,14 +162,17 @@ func (o *ClusterUninstaller) Run() error {
 		logger:  o.Logger,
 	}
 
-	deleted, err := terminateEC2InstancesByTags(ec2.New(awsSession), iamClient, o.Filters, o.Logger)
+	deleted := destroyedResources{}
+
+	resourceChanges, err := terminateEC2InstancesByTags(ctx, ec2.New(awsSession), iamClient, o.Filters, o.Logger)
+	deleted.merge(resourceChanges)
 	if err != nil {
-		return err
+		return deleted.extractRemaining(), err
```

Comment:
This is going to be a very limited view of remaining ARNs, since it does not include all the other resource types we check in the poll loop below.  How thorough do callers expect our remaining-ARN list to be?

Reply:
That is a good point. I am not sure what else we can do here because this will happen when the context expires. It may be better to collect all of the ARNs from tags upfront before attempting to delete anything. That still would not cover sub-resources, but at least it will give a more complete picture.

Reply:
> That still would not cover sub-resources...

Personally, I'm fine treating sub-resources of owned resources as also owned, so a caller who received a set of top-level resources could recurse through sub-resources and remove the whole set.  @abhinavdahiya has pushed back on that interpretation, although I'm not clear on his counter-argument.  In the context of this PR, that means that I'm fine with returning a list of whatever the tag-search logic turns up, but that you might want to run it by him before coding anything up.